src/main/java/com/elastic/support/diagnostics/DiagnosticInputs.java

```diff
@@ -241,13 +253,24 @@ public boolean runInteractive() {
 
     }
 
+    /**
+    * Elasticsearch, Kibana and Logstash have default port, on this function we set the default for Kibana or Logstash in case the port is not defined during the execution
+    * port variable, is a public int defined in the parent class to 9200 if not defined during the execution
+    *
+    * @param  String val , type of diagnostic
+    *
+    */
     public List<String> setDefaultPortForDiagType(String val) {
         // Check the diag type and reset the default port value if
         // it is a Logstash diag.
         if (val.toLowerCase().contains("logstash")) {
             if (port == 9200) {
                 port = Constants.LOGSTASH_PORT;
             }
+        } else if (val.toLowerCase().contains("kibana")) {
+            if (port == 9200) {
+                port = Constants.KIBANA_PORT;
+            }
         }
         return null;
```

Comment:
here some spaghetti, so the parent function is expecting null as response, so they will use the default. but the default is set/updated within the function that returns null. (the port we change is a public variable defined on the parent class ElasticRestClientInputs)
In a few words if we do not return null the default is not applied/updated, and we think there is an error/problem with the port defined.

But I'm not finished yet, the function is also defined in BaseInputs (parent of ElasticRestClientInputs that its parent of DiagnosticInputs) 
In BaseInputs the function really return a list, and null if the customer has set a correct port, the list returned was in case the port defined was incorrect (val < 1 || val > 65535), So it make sense in BaseInputs, but not anymore in DiagnosticInputs (as we always return null).

I was checking and all the other validate functions on the DiagnosticInputs class worked the same.
I'm not sure if the main idea was some kind of Factory design and then by mistake we add global variables...